code/game/mecha/equipment/tools/tools.dm

```diff
@@ -139,90 +173,41 @@
-		if(do_after_cooldown(target))
-			if(T == chassis.loc && src == chassis.selected)
-				if(istype(target, /turf/simulated/wall/r_wall))
-					if(do_after_cooldown(target))//To slow down how fast mechs can drill through the station
-						log_message("Drilled through [target]")
-						target.ex_act(3)
-				else if(istype(target, /turf/unsimulated/mineral))
-					for(var/turf/unsimulated/mineral/M in range(chassis,1))
-						if(get_dir(chassis,M)&chassis.dir)
-							M.GetDrilled()
-					log_message("Drilled through [target]")
-					if(locate(/obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp) in chassis.equipment)
-						var/obj/structure/ore_box/ore_box = locate(/obj/structure/ore_box) in chassis:cargo
-						if(ore_box)
-							for(var/obj/item/weapon/ore/ore in range(chassis,1))
-								if(get_dir(chassis,ore)&chassis.dir && ore.material)
-									ore_box.materials.addAmount(ore.material,1)
-									qdel(ore)
-				else if(istype(target,/turf/unsimulated/floor/asteroid))
-					for(var/turf/unsimulated/floor/asteroid/M in range(target,1))
-						M.gets_dug()
-					log_message("Drilled through [target]")
-					if(locate(/obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp) in chassis.equipment)
-						var/obj/structure/ore_box/ore_box = locate(/obj/structure/ore_box) in chassis:cargo
-						if(ore_box)
-							for(var/obj/item/weapon/ore/ore in range(target,1))
-								if(ore.material)
-									ore_box.materials.addAmount(ore.material,1)
-									qdel(ore)
-				else if(target.loc == C)
-					if(istype(target, /mob/living))
-						var/mob/living/M = target
-						M.attack_log +="\[[time_stamp()]\]<font color='orange'> Mech Drilled by [chassis.occupant.name] ([chassis.occupant.ckey]) with [src.name]</font>"
-						chassis.occupant.attack_log += "\[[time_stamp()]\]<font color='red'> Mech Drilled [M.name] ([M.ckey]) with [src.name]</font>"
-						log_attack("<font color='red'>[chassis.occupant.name] ([chassis.occupant.ckey]) mech drilled [M.name] ([M.ckey]) with [src.name]</font>" )
-						if(!iscarbon(chassis.occupant))
-							M.LAssailant = null
-						else
-							M.LAssailant = chassis.occupant
-					log_message("Drilled through [target]")
-					target.ex_act(2)
-		return 1
-
-	can_attach(obj/mecha/M as obj)
-		if(..())
-			if(istype(M, /obj/mecha/working) || istype(M, /obj/mecha/combat))
-				return 1
-		return 0
+	//OBJECT
+	//ORIENTED
+	//PROGRAMMING
 
 /obj/item/mecha_parts/mecha_equipment/tool/extinguisher
-	name = "Extinguisher"
-	desc = "Exosuit-mounted extinguisher (Can be attached to: Engineering exosuits)"
+	name = "Exosuit-Mounted Foam Extinguisher"
+	desc = "A fire extinguisher module for an exosuit. (Can be attached to: Firefighting exosuits)"
 	icon_state = "mecha_exting"
-	equip_cooldown = 5
+	origin_tech = "materials=1;engineering=2"
+	equip_cooldown = 15
 	energy_drain = 0
 	range = MELEE|RANGED
 
+	can_attach(obj/mecha/working/ripley/firefighter/M as obj)

```

Comment:
Oh my god yes I fucking hate that, ALL of Mechacode is like that and it's ugly as sin to look at.
I did realize it was bad when making it but left it like that to at least keep consistency with the rest of the file.